src/Repository/IdVerificationRepository.php

```diff
@@ -0,0 +1,95 @@
+ * @method IdVerification[]    findAll()
+ * @method IdVerification[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+ */
+class IdVerificationRepository extends ServiceEntityRepository
+{
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, IdVerification::class);
+    }
+
+    public function getOnsiteIdVerifications($site, $params): array
+    {
+        $queryBuilder = $this->createQueryBuilder('idv')
+            ->select('idv.createdTs, idv.participantId, idv.verificationType, idv.visitType, 
+                u.email, idvi.id as importId')
+            ->leftJoin('idv.user', 'u')
+            ->leftJoin('idv.import', 'idvi')
+            ->where('idv.site =:site');
+
+        if (!empty($params['participantId'])) {
+            $queryBuilder->andWhere('idv.participantId = :participantId')
+                ->setParameter('participantId', $params['participantId']);
+        }
+
+        if (!empty($params['startDate'])) {
+            $queryBuilder->andWhere('idv.createdTs >= :startDate')
+                ->setParameter('startDate', $params['startDate']);
+        }
+
+        if (!empty($params['endDate'])) {
+            $queryBuilder->andWhere('idv.createdTs <= :endDate')
+                ->setParameter('endDate', $params['endDate']->modify('+1 day'));
+        }
+
+        $queryBuilder->setParameter('site', $site);
+
+        if (isset($params['sortColumn'])) {
+            $queryBuilder->orderBy($params['sortColumn'], $params['sortDir']);
+        } else {
+            $queryBuilder->orderBy('idv.createdTs', 'DESC');
+        }
+
+        if (isset($params['start'])) {
+            return $queryBuilder
+                ->getQuery()
+                ->setFirstResult($params['start'])
+                ->setMaxResults($params['length'])
+                ->getResult();
+        } else {
+            return $queryBuilder
+                ->getQuery()
+                ->getResult();
+        }
+    }
+
+    public function getOnsiteIdVerificationsCount($site, $params): int
+    {
+        $queryBuilder = $this->createQueryBuilder('idv')
+            ->select('count(idv.id)')
+            ->leftJoin('idv.user', 'u')
+            ->leftJoin('idv.import', 'idvi')
+            ->where('idv.site =:site');
+        if (!empty($params['participantId'])) {
+            $queryBuilder->andWhere('idv.participantId = :participantId')
+                ->setParameter('participantId', $params['participantId']);
+        }
+
+        if (!empty($params['startDate'])) {
+            $queryBuilder->andWhere('idv.createdTs >= :startDate')
+                ->setParameter('startDate', $params['startDate']);
+        }
+
+        if (!empty($params['endDate'])) {
+            $queryBuilder->andWhere('idv.createdTs <= :endDate')
+                ->setParameter('endDate', $params['endDate']->modify('+1 day'));
+        }
+
+        $queryBuilder->setParameter('site', $site);
```

Comment:
Since this part of the query builder for the count is the same as the repository method above for getting the rows, could you move this into a shared private method used by both repository methods?